### DISCUSSION TITLE
Defining store with id as first parameter instead of inside options breaks WebStorm Intellisense in Vue templates

### DISCUSSION THREAD

**Discussion by mousesama** [vuejs/pinia#774] [Help and Questions]
### Reproduction

Using WebStorm IDE, using `defineStore()` function to define a store using id as the first parameter and options object as second makes WebStorm not see any of the internal properties, getters, or actions in Intellisense when used in vue templates, it only sees the internal store values. In the script section, everything works as intended. Moving id into the options object makes Intellisense behave as expected.

### Steps to reproduce the behavior

1. Define store using 
```ts
export const useMyStore = defineStore('myStore', {
    state: () => ({
        myVal: "value",
    }),
})
```
2. Export store in script of vue component to use in template
```ts
export default defineComponent({
  name: "myComponent",
  setup() {
    const myStore = useMyStore()

    return {
      myStore
    }
  }
})
```
3. Try to use the store in a template eg. ` <h1> {{ myStore.myVal }} </h1>`
4. WebStorm will complain that myVal does not exist in the object and Intellisense/autocomplete on myStore only results in showing internal store values (such as $patch)

### Expected behavior

WebStorm will see and recognize my defined values of the store alongside the internal ones.

### Actual behavior

WebStorm only sees internal store values

### Additional information

I verified the problem exists using the latest Pinia version when using the new defineStore syntax. Moving the id back to the options menu as in the old syntax does not suffer from this problem.
See discussion: https://github.com/posva/pinia/discussions/531

**Comment by posva** [comment]
I don't use WebStorm so I won't be able to help with this but I doubt there is anything to do on Pinia's code as it's just TypeScript. Maybe WebStorm is not using an up to date version of TS?

**Comment by mousesama** [comment]
Thanks for the replyt. I tried updating to the latest version of WebStorm but that did not fix it. I will try to raise the issue with JetBrains directly.

**Reply by aparajita** [reply]
I'm seeing similar behavior. What's really weird is WebStorm is correctly inferring the type of `state` as the result of `defineStore()`, but when I import the store into a component, the `state` typing gets lost.

**Comment by aparajita** [comment]


**Reply by mousesama** [reply]
I haven't tried this workaround but as I wrote above, a much simpler workaround is to simply use the old syntax of defining a store.
Instead of using 
```ts
defineStore('myStore', {
  // Stuff here
})
```
you can use
```ts
defineStore({
    id: 'myStore'
    // Stuff here
})
```
and everything will work correctly as it should.

**Reply by aparajita** [reply]
I missed that, thank you! 😁

**Reply by JackEdwardLyons** [reply]
This worked for me, in VS Code, but the docs don't explain that this is a valid option.

**Comment by joshkrz** [comment]
Thanks, this was one of the reasons autocompletion wasn't working in PHPStorm / WebStorm. I also had to export `defineStore` by default and not destructure it when importing it. I mentoned more here: https://github.com/posva/pinia/discussions/531#discussioncomment-1698659

**Reply by intermundos** [reply]
Confirming that. 

Moving store to export default enables autocompletion.

**Comment by kawazoe** [comment]
I have noticed a whole bunch of issues with types in WebStorm. This is really confusing as they are using the TypeScript Language Service to get their type information. I have never seen any issue with Angular. This stuff is only happening in vue projects. Here's an example of one of the bugs I have reported: https://youtrack.jetbrains.com/issue/WEB-52619

I recommend you create an issue in their tracker as well. Maybe they'll finally work on a fix if enough people complain...
